### PR TITLE
[EPMRPP-33139] Added react-grid-layout and buttons

### PR DESCRIPTION
**Pull request by PavelZaletski**

### CODE REVIEW

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  };
+
+  toggleFullscreen = () => {
+    this.setState({ isFullscreen: !this.state.isFullscreen });
+  };
+
+  render() {
+    const { formatMessage } = this.props.intl;
+    return (
+      <div className={`${cx('dashboard-item')}`}>
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: className={cx('dashboard-item')}

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  },
+});
+
+@connect((state) => ({
+  url: `/api/v1/${activeProjectSelector(state)}/dashboard`,
+}))
+@injectIntl
+export class DashboardItemPage extends Component {
+  static propTypes = {
+    url: PropTypes.string.isRequired,
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: Is it used here?

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      data: {
+        updateWidgets: widgets,
+      },
+    });
+  }
+
+  fetchWidgets() {
+    this.setState({ isFetching: true });
+
+    return fetch(this.props.url).then((result) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: I suggest fetching current dashboard by id and taking widgets from the response.

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    let Items = null;
+    let height = 0; // we need to set large height to avoid double scroll
+    if (widgets.length) {
+      Items = widgets.map(({ widgetPosition: [x, y], widgetSize: [w, h], widgetId }) => {
+        height += h * (rowHeight + 20);
+        return <div key={widgetId} className={cx('gadget-view')} data-grid={{ x, y, w, h }} />;
+      });
+    }
+
+    const breakpoints = { lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 };
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: Please move static objects outside of the `render`

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      this.setState({
+        widgets: currentDashboard.widgets,
+        isFetching: false,
+      });
+    });
+  }
+
+  render() {
+    const { widgets } = this.state;
+    const rowHeight = 63;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: Move constant outside of the component

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    dashboardId: '',
+  };
+
+  state = {
+    widgets: [],
+    isFetching: false,
+  };
+
+  componentDidMount() {
+    this.fetchWidgets();
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: Please also add fetching on `dashboardId` prop change

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx (hunk 1 above)
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx (hunk 2 above)
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: No. Removed

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx (hunk 3 above)
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: Implemented

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx (hunk 5 above)
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: done

**Comment by ArtsiomYeliseyenka** [comment]
Please fix Prettier errors

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+import { Widget } from './widget';
+
+const cx = classNames.bind(styles);
+const ResponsiveGridLayout = WidthProvider(Responsive);
+const rowHeight = 63;
+const breakpoints = { lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 };
+const cols = { lg: 12, md: 12, sm: 4, xs: 4, xxs: 4 };
+
+@connect((state) => ({
+  url: `/api/v1/${activeProjectSelector(state)}/dashboard`,
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]:
```
@connect((state, ownProps) => ({
  url: `/api/v1/${activeProjectSelector(state)}/dashboard/${ownProps.dashboardId}`
}))
```

Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: done

**Pull request merged**